cpp/log/cert.cc

```diff
@@ -671,6 +696,31 @@ util::Status Cert::PublicKeySha256Digest(string* result) const {
 }
 
 
+StatusOr<string> Cert::SPKI() const {
+  if (!IsLoaded()) {
+    LOG(ERROR) << "Cert not loaded";
+    return util::Status(Code::FAILED_PRECONDITION, "Cert not loaded");
+  }
+
+  unsigned char* der_buf(nullptr);
+  int der_length =
+      i2d_X509_PUBKEY(X509_get_X509_PUBKEY(x509_.get()), &der_buf);
+  if (der_length < 0) {
+    // What does this return value mean? Let's assume it means the cert
+    // is bad until proven otherwise.
+    LOG(WARNING) << "Failed to serialize the Subject Public Key Info";
+    LOG_OPENSSL_ERRORS(WARNING);
+    return util::Status(Code::INVALID_ARGUMENT, "SPKI() failed");
+  }
+
+  string result;
+  result.assign(string(reinterpret_cast<char*>(der_buf), der_length));
+
+  OPENSSL_free(der_buf);
+  return result;
+}
+
+
 util::Status Cert::SPKISha256Digest(string* result) const {
```

Comment:
I'll add a bit of test coverage for the (currently entirely untested) SPKISha256Digest first.